File: backend/routes/claimRoutes.js
// backend/routes/claimRoutes.js
const express = require('express');
const router = express.Router();
const { claimOffer, getUserClaims } = require('../controllers/claimController');
const verifyToken = require('../middleware/authMiddleware'); // Bring in the bouncer!

// Fetch a user's claims
router.get('/my-claims', verifyToken, getUserClaims);

// POST /api/claims/:offer_id 
// (The :offer_id is a dynamic parameter so users can claim different offers)
router.post('/:offer_id', verifyToken, claimOffer);

module.exports = router;

File: backend-python/routes/claim_routes.py
# backend-python/routes/claim_routes.py
from fastapi import APIRouter, Depends

from controllers import claim_controller as ctrl
from middleware.auth import verify_token

router = APIRouter()


@router.get("/my-claims")
async def get_user_claims(current_user: dict = Depends(verify_token)):
    return await ctrl.get_user_claims(current_user["id"])


@router.post("/{offer_id}", status_code=201)
async def claim_offer(offer_id: int, current_user: dict = Depends(verify_token)):
    return await ctrl.claim_offer(offer_id, current_user["id"], current_user.get("role", ""))
